.pre-commit-config.yaml

```diff
@@ -339,6 +339,11 @@ repos:
         language: python
         entry: python scripts/validate_unwanted_patterns.py --validation-type="strings_with_wrong_placed_whitespace"
         types_or: [python, cython]
+    -   id: unwanted-patterns-nodefault-not-used-for-typing
+        name: Check for `pandas._libs.lib.NoDefault` not used for typing
```

Comment:
Just to clarify `pandas._libs.lib.NoDefault` should **only** be used for typing

Reply:
"Check for `pandas._libs.lib.NoDefault` is only used for typing"

Reply:
Yeah the script looks correct, just a suggestion that this phrasing was a little unclear

Reply:
If we're going to rename this, can we rename it everywhere please? there's a few places where it still says "not used in typing" - let's use consistent wording, and for the error message, let's say "used not only in typing"